From 00e617a432a2e968cc149995758c7a949441f65b Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Sun, 8 Sep 2013 15:14:13 -0400
Subject: [PATCH 02/20] Refactor/simplify alternate auth file open handling.

---
 cloudprint/cloudprint.py | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 614c501..6a3dacc 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -428,16 +428,12 @@ def main():
 
     # Check if authentification is needed
     if not cpp.get_saved_auth():
-        if authfile:
-            if os.path.exists(authfile):
-                account_file = open(authfile)
-                cpp.username = account_file.next().rstrip()
-                cpp.password = account_file.next().rstrip()
-                account_file.close()
-
-            else:
-                cpp.username = raw_input('Google username: ')
-                cpp.password = getpass.getpass()
+        if authfile and os.path.exists(authfile):
+            account_file = open(authfile)
+            cpp.username = account_file.next().rstrip()
+            cpp.password = account_file.next().rstrip()
+            account_file.close()
+
         else:
           cpp.username = raw_input('Google username: ')
           cpp.password = getpass.getpass()
-- 
1.8.4.rc3

