From: Stephen Blott <smblott+github@gmail.com>
Date: Tue, 21 Jan 2014 10:11:08 +0000
Subject: Remove debug code.

---
 cloudprint/cloudprint.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 2903bbf..87db52f 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -307,9 +307,6 @@ def sync_printers(cups_connection, cpp):
     local_printer_names = set([ prn for prn in local_printer_names if     match_re(prn,cpp.include,True) ])
     local_printer_names = set([ prn for prn in local_printer_names if not match_re(prn,cpp.exclude     ) ])
 
-    print local_printer_names
-    sys.exit(0)
-
     #New printers
     for printer_name in local_printer_names - remote_printer_names:
         try:
