From 6d0e9f51fe90eb625833d7f79113f5277c49a0d9 Mon Sep 17 00:00:00 2001
From: Richard van der Hoff <git@rvanderhoff.org.uk>
Date: Mon, 25 Mar 2013 23:02:09 +0000
Subject: [PATCH 08/20] fix 'invalid argument' errors

---
 cloudprint/xmpp.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cloudprint/xmpp.py b/cloudprint/xmpp.py
index e682c4c..35b5934 100644
--- a/cloudprint/xmpp.py
+++ b/cloudprint/xmpp.py
@@ -176,6 +176,9 @@ class XmppConnection(object):
                     waittime = remaining
                     self._LOGGER.debug("%f seconds until timeout" % waittime)
 
+            if waittime < 0:
+                waittime = 0
+                    
             sock = self._xmppsock
             (r, w, e) = select.select([sock], [], [sock], waittime)
 
-- 
1.8.4.rc3

