From: Stephen Blott <smblott+github@gmail.com>
Date: Tue, 21 Jan 2014 10:24:26 +0000
Subject: Use -x from exclude.

---
 README.rst               | 4 ++--
 cloudprint/cloudprint.py | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/README.rst b/README.rst
index 0f50acc..4fd567b 100644
--- a/README.rst
+++ b/README.rst
@@ -21,7 +21,7 @@ Usage
   -c              : establish and store login credentials, then exit
   -f              : 'fast poll', if notifications aren't working
   -i regexp       : include files matching regexp
-  -e regexp       : exclude filees matching regexp
+  -x regexp       : exclude filees matching regexp
                     regexp: a Python regexp, which is matched against the
                             start of the printer name
   -h              : display this help
@@ -50,7 +50,7 @@ Include only the printers "`lp`" and "`2up`":
 Exclude all printers whose names start with "`GCP-`":
 ::
 
-  cloudprint -e GCP-
+  cloudprint -x GCP-
 
 By default, all printers are included.  For the include and exclude options,
 the argument is a regular expression which is matched against the start of the
diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 87db52f..6e024ea 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -402,13 +402,13 @@ def usage():
     print '-c\t\t: establish and store login credentials, then exit'
     print '-f\t\t: use fast poll if notifications are not working'
     print '-i regexp\t: include local printers matching regexp'
-    print '-e regexp\t: exclude local printers matching regexp'
+    print '-x regexp\t: exclude local printers matching regexp'
     print '\t\t regexp: a Python regexp, which is matched against the start of the printer name'
     print '-v\t\t: verbose logging'
     print '-h\t\t: display this help'
 
 def main():
-    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:cvfi:e:')
+    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:cvfi:x:')
     daemon = False
     logout = False
     pidfile = None
@@ -437,7 +437,7 @@ def main():
             fastpoll = True
         elif o == '-i':
             include.append(a)
-        elif o == '-e':
+        elif o == '-x':
             exclude.append(a)
         elif o =='-h':
             usage()
