From: David Steele <dsteele@gmail.com>
Date: Sun, 5 Jul 2015 12:39:40 -0400
Subject: Spelling - 'claimed'

---
 cloudprint/cloudprint.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index c0d6b73..399a6a6 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -127,7 +127,7 @@ class CloudPrintAuth(object):
 
         self.xmpp_jid = poll['xmpp_jid']
         self.email = poll['user_email']
-        print 'Printer clammed by {0}.'.format(self.email)
+        print 'Printer claimed by {0}.'.format(self.email)
 
         token = requests.post(
             'https://accounts.google.com/o/oauth2/token',
