From: Dean Montgomery <dmonty@sd73.bc.ca>
Date: Tue, 29 Sep 2015 15:04:38 -0700
Subject: Add site-name prefix for multi-site support

---
 README.rst               |  1 +
 cloudprint/cloudprint.py | 20 +++++++++++++++++++-
 2 files changed, 20 insertions(+), 1 deletion(-)

diff --git a/README.rst b/README.rst
index 93f3410..e99c40e 100644
--- a/README.rst
+++ b/README.rst
@@ -28,6 +28,7 @@ Usage
   -x regexp       : exclude filees matching regexp
                     regexp: a Python regexp, which is matched against the
                             start of the printer name
+  -s sitename     : one-word site-name that will prefix printers
   -h              : display this help
 
 Google accounts with 2 step verification enabled need to use an
diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 9873c64..dd4b8c0 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -210,6 +210,7 @@ class CloudPrintProxy(object):
     def __init__(self, auth):
         self.auth = auth
         self.sleeptime = 0
+        self.site = ''
         self.include = []
         self.exclude = []
 
@@ -222,7 +223,11 @@ class CloudPrintProxy(object):
             },
         ).json()
         return [
-            PrinterProxy(self, p['id'], p['name'])
+            PrinterProxy(
+                self,
+                p['id'],
+                re.sub('^' + self.site + '-', '', p['name'])
+            )
             for p in printers['printers']
         ]
 
@@ -237,6 +242,8 @@ class CloudPrintProxy(object):
         LOGGER.debug('Deleted printer ' + printer_id)
 
     def add_printer(self, name, description, ppd):
+        if self.site:
+            name = self.site + '-' + name
         self.auth.session.post(
             PRINT_CLOUD_URL + 'register',
             {
@@ -253,6 +260,8 @@ class CloudPrintProxy(object):
         LOGGER.debug('Added Printer ' + name)
 
     def update_printer(self, printer_id, name, description, ppd):
+        if self.site:
+            name = self.site + '-' + name
         self.auth.session.post(
             PRINT_CLOUD_URL + 'update',
             {
@@ -522,6 +531,14 @@ def parse_args():
         '--syslog-address',
         help='syslog address to use in daemon mode',
     )
+    parser.add_argument(
+        '-s',
+        metavar='sitename',
+        dest='site',
+        default='',
+        help='one-word site-name that will prefix printers',
+    )
+
     return parser.parse_args()
 
 
@@ -576,6 +593,7 @@ def main():
 
     cpp.include = args.include
     cpp.exclude = args.exclude
+    cpp.site = args.site
 
     printers = list(cups_connection.getPrinters().keys())
     if not printers:
