From: Stephen Blott <smblott+github@gmail.com>
Date: Wed, 22 Jan 2014 08:41:29 +0000
Subject: Catch exception for invalid regular expressions.

---
 cloudprint/cloudprint.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 36982a3..2a546f2 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -294,7 +294,11 @@ class App(object):
 #True if printer name matches *any* of the regular expressions in regexps
 def match_re(prn, regexps, empty=False):
     if len(regexps):
-        return re.match(regexps[0], prn, re.UNICODE) or match_re(prn, regexps[1:])
+        try:
+           return re.match(regexps[0], prn, re.UNICODE) or match_re(prn, regexps[1:])
+        except:
+           sys.stderr.write('cloudprint: invalid regular expression: ' + regexps[0] + '\n')
+           sys.exit(1)
     else:
         return empty
 
