From 7a6b98ef5035663c5d8c20307d3142317d960fe3 Mon Sep 17 00:00:00 2001
From: Stephen Blott <smblott+github@gmail.com>
Date: Tue, 21 Jan 2014 10:24:26 +0000
Subject: [PATCH 10/12] Use -x from exclude.

---
 README.rst               | 4 ++--
 cloudprint/cloudprint.py | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/README.rst b/README.rst
index fb23eb7..40bbe73 100644
--- a/README.rst
+++ b/README.rst
@@ -21,7 +21,7 @@ Usage
   -u              : store username/password in addition to login token
                     to avoid authentication expiration
   -i regexp       : include files matching regexp
-  -e regexp       : exclude filees matching regexp
+  -x regexp       : exclude filees matching regexp
                     regexp: a Python regexp, which is matched against the
                             start of the printer name
   -h              : display this help
@@ -50,7 +50,7 @@ Include only the printers "`lp`" and "`2up`":
 Exclude all printers whose names start with "`GCP-`":
 ::
 
-  cloudprint -e GCP-
+  cloudprint -x GCP-
 
 By default, all printers are included.  For the include and exclude options,
 the argument is a regular expression which is matched against the start of the
diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 941931a..bf31425 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -416,13 +416,13 @@ def usage():
     print '-f\t\t: use fast poll if notifications are not working'
     print '-u\t\t: store username/password in addition to login token'
     print '-i regexp\t: include local printers matching regexp'
-    print '-e regexp\t: exclude local printers matching regexp'
+    print '-x regexp\t: exclude local printers matching regexp'
     print '\t\t regexp: a Python regexp, which is matched against the start of the printer name'
     print '-v\t\t: verbose logging'
     print '-h\t\t: display this help'
 
 def main():
-    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:cuvfi:e:')
+    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:cuvfi:x:')
     daemon = False
     logout = False
     pidfile = None
@@ -454,7 +454,7 @@ def main():
             storepw = True
         elif o == '-i':
             include.append(a)
-        elif o == '-e':
+        elif o == '-x':
             exclude.append(a)
         elif o =='-h':
             usage()
-- 
1.8.5.2

