From a4537dcee7e9fd859d719bdf1ae58f41c3acfe1d Mon Sep 17 00:00:00 2001
From: Stephen Blott <smblott+github@gmail.com>
Date: Wed, 22 Jan 2014 07:17:45 +0000
Subject: [PATCH 11/12] Remove whitespace, response to comment from @armooo.

---
 cloudprint/cloudprint.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index bf31425..797bca4 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -310,8 +310,8 @@ def sync_printers(cups_connection, cpp):
     remote_printer_names = set(remote_printers)
 
     #Include/exclude local printers
-    local_printer_names = set([ prn for prn in local_printer_names if     match_re(prn,cpp.include,True) ])
-    local_printer_names = set([ prn for prn in local_printer_names if not match_re(prn,cpp.exclude     ) ])
+    local_printer_names = set([ prn for prn in local_printer_names if match_re(prn,cpp.include,True) ])
+    local_printer_names = set([ prn for prn in local_printer_names if not match_re(prn,cpp.exclude ) ])
 
     #New printers
     for printer_name in local_printer_names - remote_printer_names:
-- 
1.8.5.2

