From: David Steele <dsteele@gmail.com>
Date: Fri, 8 Nov 2013 09:37:33 -0500
Subject: cloudprint.py: Log when get_rest() changes token

---
 cloudprint/cloudprint.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 2a546f2..0163837 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -143,6 +143,7 @@ class CloudPrintProxy(object):
                 def f(*arg, **karg):
                     r = attr(*arg, **karg)
                     if 'update-client-auth' in r.headers:
+                        LOGGER.info("Updating authentication token")
                         self.set_auth(r.headers['update-client-auth'])
                     return r
                 return f
