From 70cd2557fc2d9662c9310bb28164ba54e9c0c782 Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Fri, 8 Nov 2013 09:37:33 -0500
Subject: [PATCH 25/29] cloudprint.py: Log when get_rest() changes token

---
 cloudprint/cloudprint.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index a84600a..38b483a 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -139,6 +139,7 @@ class CloudPrintProxy(object):
                 def f(*arg, **karg):
                     r = attr(*arg, **karg)
                     if 'update-client-auth' in r.headers:
+                        LOGGER.info("Updating authentication token")
                         self.set_auth(r.headers['update-client-auth'])
                     return r
                 return f
-- 
1.8.4.4

