From 554be3b53923501269c10a1e2d382b38b23f6095 Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Sat, 7 Dec 2013 12:35:24 -0500
Subject: [PATCH 30/31] Remove unused printer_id member in CloudPrintProxy

---
 cloudprint/cloudprint.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index ab0e70a..00e071f 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -47,7 +47,6 @@ class CloudPrintProxy(object):
     def __init__(self, verbose=True):
         self.verbose = verbose
         self.auth = None
-        self.printer_id = None
         self.cups= cups.Connection()
         self.proxy =  platform.node() + '-Armooo-PrintProxy'
         self.auth_path = os.path.expanduser('~/.cloudprintauth')
-- 
1.8.5.1

